src/main/java/com/firebolt/jdbc/type/array/SqlArrayUtil.java

```diff
@@ -101,11 +98,9 @@ private static String getArraySeparator(int dimension) {
 		return stringBuilder.toString();
 	}
 
+	@SuppressWarnings("java:S5852") // simplicity of code that uses regex here is more important than pre-mature optimisation
 	private static String removeQuotesAndTransformNull(String s) {
-		if (StringUtils.equals(s, "NULL")) {
-			return "\\N";
-		}
-		return CharMatcher.is('\'').trimFrom(s);
+		return "NULL".equals(s) ? "\\N" : s.replaceFirst("^'+", "").replaceFirst("'+$", "");
```

Comment:
I believe `S5852` is concerned about the DOS vulnerability of this regex rather than optimisation. Are we safe here?

Reply:
@ptiurin, you are right. I was astonished as well. The explanation here is that regex generally can cause performance issues. However in fact I do not think that this is real problem here.